modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -174,6 +184,9 @@ public void run(Context context) {
 
         try {
             exec();
+            if (context.isDebugEnabled()) {
+                VMDebugManager.getInstance().setDone(true);
```

Comment:
we were planning to make the main thread blocking only right? in that case this should be fine, because in service scenarios, this won't have an effect